cairo/ethereum/cancun/fork.cairo

```diff
@@ -732,3 +754,66 @@ func _check_versioned_hashes_version{range_check_ptr}(
 
     return ();
 }
+
+func get_last_256_block_hashes{
+    range_check_ptr, bitwise_ptr: BitwiseBuiltin*, keccak_ptr: KeccakBuiltin*
+}(chain: BlockChain) -> ListHash32 {
+    alloc_locals;
+
+    // If no blocks, return empty array
+    if (chain.value.blocks.value.len == 0) {
+        let (empty_hashes_alloc: Hash32*) = alloc();
+        tempvar empty_hashes = ListHash32(new ListHash32Struct(data=empty_hashes_alloc, len=0));
+        return empty_hashes;
+    }
+
+    // Get last 255 blocks or all blocks if less than 255
+    let less_than_255 = is_le_felt(chain.value.blocks.value.len, 255);
+    if (less_than_255 != FALSE) {
+        tempvar start_idx = 0;
+    } else {
+        tempvar start_idx = chain.value.blocks.value.len - 255;
+    }
+    tempvar recent_blocks_len = chain.value.blocks.value.len - start_idx;
+
+    // Allocate list for hashes
+    let (hashes: Hash32*) = alloc();
+
+    // Get parent hashes from recent blocks
+    _get_parent_hashes{hashes=hashes}(
+        chain.value.blocks.value.data + start_idx, recent_blocks_len, 0
+    );
+
+    // Add hash of most recent block
+    let most_recent_block: Block* = chain.value.blocks.value.data + chain.value.blocks.value.len -
+        1;
+    let most_recent_hash = keccak256_header(most_recent_block.value.header);
+    assert hashes[recent_blocks_len] = most_recent_hash;
+    tempvar list_hash_32 = ListHash32(new ListHash32Struct(data=hashes, len=recent_blocks_len + 1));
+    return list_hash_32;
+}
+
+// Helper function to recursively get parent hashes
+func _get_parent_hashes{range_check_ptr, hashes: Hash32*}(blocks: Block*, len: felt, idx: felt) {
+    if (idx == len) {
+        return ();
+    }
+
+    // Get block at current index
+    let block: Block* = blocks + idx;
+    // Store parent hash
+    assert hashes[idx] = block.value.header.value.parent_hash;
+
+    return _get_parent_hashes(blocks, len, idx + 1);
```

Comment:
considering that there's no inner function call this could be refactored to a `loop` easily